pkg/client/client.go

```diff
@@ -34,13 +34,15 @@ type Client struct {
 	deviceID     uint64
 	electionID   p4_v1.Uint128
 	p4Info       *p4_config_v1.P4Info
+	role         *p4_v1.Role
 	streamSendCh chan *p4_v1.StreamMessageRequest
 }
 
 func NewClient(
```

Comment:
can we write `NewClientForRole` here, and leave the current `NewClient` untouched? In this case, it's an easy way not to break any consumers of this module.

Reply:
Makes sense. Let me fix this in the next patch.